### ISSUE TITLE
How to disable Ctrl+click default behavior

### ISSUE THREAD

**Issue by scoutw2p** [ThibaultJanBeyer/DragSelect#20]
I am using the jQuery.contextmenu library and am running into issues when ctrl+clicking (Mac). DragSelect seems to not respect the ctrl+click as a right click and is behaving as if I just left clicked the element, and clears my entire selection except for the clicked element. I have removed all multiSelect modifier keys from the init and the issue persists. 

Any suggestions?

Sam

**Comment by ThibaultJanBeyer** [comment]
Hey @scoutw2p, thanks for reporting your issue.

Indeed DragSelect only checks for real right-clicks with the `isRightClick` method. <kbd>ctrl</kbd>+click is a fake right-click which was only implemented by Apple as a workaround for their weird ancient single-click mouses. In my opinion, implementing a <kbd>ctrl</kbd> check for right click in the tool itself is a bad idea since not every user is a mac user and this could potentially break the code on Linux and Windows. Also, checking for a specific operating system is a bad practice since it is not guaranteed to work and very fiddly.

**TL;DR** I’m here to help. So if you still want to prevent the selection on <kbd>ctrl</kbd>+click, following choices come to my mind:
## The clean way:
(using the `onDragStart` callback)
```
var ds = new DragSelect({
    ...
    multiSelectKeys: [],
    onDragStart: function(event) {

      if(event["ctrlKey"]) {  // your custom check for "ctrl-rightclick"
        this.break();  // stops the execution of further DragSelect code
      }

    }
});
```
## The ugly way:
(overwriting `isRightClick` with your own method)
```
var ds = new DragSelect(...);
ds.isRightClick = function(event) {

    var isRightMB = DragSelect.prototype.isRightClick.call(this, event);

    if(event["ctrlKey"]) {   // your custom check for "ctrl-rightclick"
        isRightMB = true;
    }

    return isRightMB;

};
```
Note: the above code examples are very quick drafts, for your specific use case, you might want to check for the operating system as well…

I hope I could help, I’m closing this for now. If you have further issues feel free to comment here or open new issues.

Thanks again for using the tool. It makes me really happy to see that DragSelect is useful for other people.
